extensions/ql-vscode/src/test-adapter.ts

```diff
@@ -352,49 +243,40 @@ export class QLTestAdapter extends DisposableObject implements TestAdapter {
     }
   }
 
-  private async runTests(
-    tests: string[],
-    cancellationToken: CancellationToken,
-  ): Promise<void> {
-    const workspacePaths = getOnDiskWorkspaceFolders();
-    for await (const event of this.cliServer.runTests(tests, workspacePaths, {
-      cancellationToken,
-      logger: testLogger,
-    })) {
-      const state = event.pass
-        ? "passed"
-        : event.messages?.length
-        ? "errored"
-        : "failed";
-      let message: string | undefined;
-      if (event.failureDescription || event.diff?.length) {
-        message =
-          event.failureStage === "RESULT"
-            ? [
-                "",
-                `${state}: ${event.test}`,
-                event.failureDescription || event.diff?.join("\n"),
-                "",
-              ].join("\n")
-            : [
-                "",
-                `${event.failureStage?.toLowerCase()} error: ${event.test}`,
-                event.failureDescription ||
-                  `${event.messages[0].severity}: ${event.messages[0].message}`,
-                "",
-              ].join("\n");
-        void testLogger.log(message);
-      }
-      this._testStates.fire({
-        type: "test",
-        state,
-        test: event.test,
-        message,
-        decorations: event.messages?.map((msg) => ({
-          line: msg.position.line,
-          message: msg.message,
-        })),
-      });
+  private async processTestEvent(event: TestCompleted): Promise<void> {
+    const state = event.pass
+      ? "passed"
+      : event.messages?.length
+      ? "errored"
+      : "failed";
+    let message: string | undefined;
+    if (event.failureDescription || event.diff?.length) {
+      message =
+        event.failureStage === "RESULT"
+          ? [
+              "",
+              `${state}: ${event.test}`,
+              event.failureDescription || event.diff?.join("\n"),
+              "",
+            ].join("\n")
+          : [
+              "",
+              `${event.failureStage?.toLowerCase()} error: ${event.test}`,
```

Comment:
This will print `undefined` error if the `failureStage` is empty. Should we printing an empty string or `unknown` instead?

Reply:
Fixed.